src/Custom/Embeddings/OpenAIEmbedding.cs

```diff
@@ -118,32 +118,36 @@ private static ReadOnlyMemory<float> ConvertToVectorOfFloats(BinaryData binaryDa
 
         // Decode base64 string to bytes.
         byte[] bytes = ArrayPool<byte>.Shared.Rent(Base64.GetMaxDecodedFromUtf8Length(base64.Length));
-        OperationStatus status = Base64.DecodeFromUtf8(base64, bytes.AsSpan(), out int bytesConsumed, out int bytesWritten);
-        if (status != OperationStatus.Done || bytesWritten % sizeof(float) != 0)
-        {
-            ThrowInvalidData();
-        }
-
-        // Interpret bytes as floats
-        float[] vector = new float[bytesWritten / sizeof(float)];
-        bytes.AsSpan(0, bytesWritten).CopyTo(MemoryMarshal.AsBytes(vector.AsSpan()));
-        if (!BitConverter.IsLittleEndian)
-        {
-            Span<int> ints = MemoryMarshal.Cast<float, int>(vector.AsSpan());
+        try
+        {
+            OperationStatus status = Base64.DecodeFromUtf8(base64, bytes.AsSpan(), out int bytesConsumed, out int bytesWritten);
+            if (status != OperationStatus.Done || bytesWritten % sizeof(float) != 0)
+            {
+                ThrowInvalidData();
+            }
+
+            // Interpret bytes as floats
+            float[] vector = new float[bytesWritten / sizeof(float)];
+            bytes.AsSpan(0, bytesWritten).CopyTo(MemoryMarshal.AsBytes(vector.AsSpan()));
+            if (!BitConverter.IsLittleEndian)
+            {
+                Span<int> ints = MemoryMarshal.Cast<float, int>(vector.AsSpan());
 #if NET8_0_OR_GREATER
-            BinaryPrimitives.ReverseEndianness(ints, ints);
+                BinaryPrimitives.ReverseEndianness(ints, ints);
 #else
             for (int i = 0; i < ints.Length; i++)
             {
                 ints[i] = BinaryPrimitives.ReverseEndianness(ints[i]);
             }
 #endif
+            }
+            return new ReadOnlyMemory<float>(vector);
+        }
+        finally
+        {
+            ArrayPool<byte>.Shared.Return(bytes);

```

Comment:
This is OK. FWIW, though, we generally don't consider it a bug to not return an ArrayPool array in exceptional situations.